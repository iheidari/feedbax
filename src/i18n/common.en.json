{
  "date-format": "MM/DD/YYYY",
  "Title": "Title",
  "Description": "Description",
  "Delete": "Delete",
  "Previous": "Previous",
  "Next": "Next",
  "Save": "Save",
  "Back": "Back",
  "Ok": "OK",
  "Cancel": "Cancel",
  "Confirmation": "Confirmation",
  "Are you sure?": "Are you sure?",
  "Deleted Successfully": "Deleted Successfully",
  "Saved Successfully": "Saved Successfully",
  "None": "None"
}
