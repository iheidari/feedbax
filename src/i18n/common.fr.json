{
  "date-format": "DD-MM-YYYY",
  "Title": "Titre",
  "Description": "Description",
  "Delete": "Supprimer",
  "Previous": "Précédent",
  "Next": "Suivant",
  "Save": "Sauvegarder",
  "Back": "Retour",
  "Ok": "D'accord",
  "Cancel": "Annuler",
  "Confirmation": "Confirmation",
  "Are you sure?": "Êtes-vous sûr?",
  "Deleted Successfully": "Supprimé avec succès",
  "Saved Successfully": "Enregistré avec succès",
  "None": "Non"
}
