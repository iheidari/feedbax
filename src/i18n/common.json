{
  "en": {
    "translation": { "welcome-to-react": "Welcome to React and react-i18next" }
  },
  "fr": {
    "translation": { "welcome-to-react": "Bienvenue Ã  React et react-i18next" }
  }
}
