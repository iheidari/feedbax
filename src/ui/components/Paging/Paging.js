import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { useTranslation } from 'react-i18next';
const showButton = (current, page, pageCount) =>
  page <= 2 ||
  page >= pageCount - 1 ||
  (page >= current - 1 && page <= current + 1);

const Paging = ({ current, take, count, onPagerChange }) => {
  const { t } = useTranslation();
  const pageCount = Math.ceil(count / take);
  let pageButtons = [];

  pageButtons.push(
    <Button key={0} onClick={current <= 1 ? null : onPagerChange('p', take)}>
      {t('Previous')}
    </Button>
  );

  for (let page = 1; page <= pageCount; page++) {
    if (showButton(current, page, pageCount)) {
      pageButtons.push(
        <Button
          key={page}
          color={page === current ? 'secondary' : 'primary'}
          onClick={page === current ? null : onPagerChange(page, take)}
        >
          {page}
        </Button>
      );
    } else {
      if (showButton(current, page - 1, pageCount))
        pageButtons.push(<span key={page}> ... </span>);
    }
  }

  pageButtons.push(
    <Button
      key={pageCount + 1 || 1}
      onClick={current >= pageCount ? null : onPagerChange('n', take)}
    >
      {t('Next')}
    </Button>
  );

  return pageButtons;
};

Paging.propTypes = {
  current: PropTypes.number,
  take: PropTypes.number,
  count: PropTypes.number,
  onPagerChange: PropTypes.func.isRequired
};

export default Paging;
