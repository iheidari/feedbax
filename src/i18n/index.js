import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import commonEn from './common.en.json';
import commonFr from './common.fr.json';

const resources = {
  en: { translation: commonEn },
  fr: { translation: commonFr }
};

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false
    }
  });

export default i18n;
