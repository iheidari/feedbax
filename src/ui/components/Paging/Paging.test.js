import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Paging from './Paging';
import i18n from '../../../i18n';

Enzyme.configure({ adapter: new Adapter() });

describe('Paging', () => {
  test('renders', () => {
    const pagingComponent = <Paging onPagerChange={() => {}} />;
    const paging = shallow(pagingComponent);
    expect(paging.exists()).toBe(true);
  });

  const onPagerChange = page => page;

  const pagingComponent = (
    <Paging current={5} take={3} count={28} onPagerChange={onPagerChange} />
  );

  test('snapshot', () => {
    const paging = shallow(pagingComponent);
    expect(paging).toMatchSnapshot();
  });

  test('normal view', () => {
    const pagingComponent = (
      <Paging current={1} take={3} count={10} onPagerChange={onPagerChange} />
    );
    const paging = shallow(pagingComponent);

    //Number of buttons Previous + 4 Pages + Next
    expect(paging.find('WithStyles(Button)').length).toBe(6);

    //Text for first button(Previous)
    expect(paging.first('WithStyles(Button)').props().children).toBe(
      'Previous'
    );

    //Text for last button(Next)
    expect(paging.last('WithStyles(Button)').props().children).toBe('Next');

    //onclick for previous button when current is 1
    expect(
      paging
        .find('WithStyles(Button)')
        .at(0)
        .props().onClick
    ).toBeNull();

    //onclick for current button
    expect(
      paging
        .find('WithStyles(Button)')
        .at(1)
        .props().onClick
    ).toBeNull();

    //color for current button (should be secondary)
    expect(
      paging
        .find('WithStyles(Button)')
        .at(1)
        .props().color
    ).toBe('secondary');

    //onclick for non-current button
    expect(
      paging
        .find('WithStyles(Button)')
        .at(2)
        .props().onClick
    ).toBe(2);

    //color for non-current button (should be primary)
    expect(
      paging
        .find('WithStyles(Button)')
        .at(2)
        .props().color
    ).toBe('primary');

    //onclick for Next button
    expect(paging.last('WithStyles(Button)').props().onClick).toBe('n');
  });

  test('full view', () => {
    const paging = shallow(pagingComponent);

    //Number of buttons Previous + 7 Pages + Next
    expect(paging.find('WithStyles(Button)').length).toBe(9);
    expect(paging.find('span').length).toBe(2);
    //span text should be ...
    //expect(paging.find('WithStyles(Button)').text()).toBe('...');

    //Text for first button(Previous)
    expect(paging.first('WithStyles(Button)').props().children).toBe(
      'Previous'
    );

    //Text for last button(Next)
    expect(paging.last('WithStyles(Button)').props().children).toBe('Next');

    //onclick for previous button when current is 1
    expect(
      paging
        .find('WithStyles(Button)')
        .at(0)
        .props().onClick
    ).toBe('p');

    //onclick for current button
    expect(
      paging
        .find('WithStyles(Button)')
        .at(4)
        .props().onClick
    ).toBeNull();

    //color for current button (should be secondary)
    expect(
      paging
        .find('WithStyles(Button)')
        .at(4)
        .props().color
    ).toBe('secondary');

    //onclick for non-current button
    expect(
      paging
        .find('WithStyles(Button)')
        .at(2)
        .props().onClick
    ).toBe(2);

    //color for non-current button (should be primary)
    expect(
      paging
        .find('WithStyles(Button)')
        .at(2)
        .props().color
    ).toBe('primary');

    //onclick for Next button
    expect(paging.last('WithStyles(Button)').props().onClick).toBe('n');
  });
});
