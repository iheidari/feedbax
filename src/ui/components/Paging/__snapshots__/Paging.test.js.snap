// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging snapshot 1`] = `
Array [
  <WithStyles(Button)
    key="0"
    onClick="p"
  >
    Previous
  </WithStyles(Button)>,
  <WithStyles(Button)
    color="primary"
    key="1"
    onClick={1}
  >
    1
  </WithStyles(Button)>,
  <WithStyles(Button)
    color="primary"
    key="2"
    onClick={2}
  >
    2
  </WithStyles(Button)>,
  <span
    key="3"
  >
     ... 
  </span>,
  <WithStyles(Button)
    color="primary"
    key="4"
    onClick={4}
  >
    4
  </WithStyles(Button)>,
  <WithStyles(Button)
    color="secondary"
    key="5"
    onClick={null}
  >
    5
  </WithStyles(Button)>,
  <WithStyles(Button)
    color="primary"
    key="6"
    onClick={6}
  >
    6
  </WithStyles(Button)>,
  <span
    key="7"
  >
     ... 
  </span>,
  <WithStyles(Button)
    color="primary"
    key="9"
    onClick={9}
  >
    9
  </WithStyles(Button)>,
  <WithStyles(Button)
    color="primary"
    key="10"
    onClick={10}
  >
    10
  </WithStyles(Button)>,
  <WithStyles(Button)
    key="11"
    onClick="n"
  >
    Next
  </WithStyles(Button)>,
]
`;
